src/detect.h

```diff
@@ -909,6 +909,8 @@ typedef struct DetectEngineThreadCtx_ {
     uint8_t *base64_decoded;
     int base64_decoded_len;
     int base64_decoded_len_max;
+    uint8_t alert_flags;
```

Comment:
I don't follow why this has to be in det_ctx. It seems it's a per-sig flag and you have the sig available at every place you need it. So it makes much more sense to me to have it in the sig.